### DISCUSSION TITLE
RuntimeError: Cannot locate slurm_version.h in /usr/include/slurm

### DISCUSSION THREAD

**Discussion by HealthyPear** [PySlurm/pyslurm#350] [Q&A] [answered]
I am working on a cluster with a Slurm 20.02.7 installation not performed by me.

I wanted to install _pyslurm_ to manage some administrative tasks I have been assigned to, but at the time of installation pip fails with the error in subject.

I see that under `/usr/include/slurm` I have these files

`pmi2.h  pmi.h  slurmdb.h  slurm_errno.h  slurm.h  smd_ns.h  spank.h`

but I cannot locate slurm_version.h anywhere.

Is this a problem related to the installed slurm version or a bug of pyslurm (or both)?

**Comment by HealthyPear** [comment]
ah I am guessing that I should install the `slurm-devel` package or similar for this...am I right?

**Comment by HealthyPear** [comment]
nevermind, it's installed  - back to square one

**Comment by schluenz** [comment]
slurm_version.h was only added in 2021: https://github.com/SchedMD/slurm/commit/f94f1793124a4d703e90f1bcd2cc435f8671cf54

**Reply by HealthyPear** [reply]
ah that explains it then....is there a way to use pyslurm on my slurm version?

**Comment by tazend** [comment, marked as answer]
Hi @HealthyPear 

as @schluenz pointed out already, slurm_version.h did not exist yet in Slurm 20.2.x. The error you are receiving might indicate that you are trying to use a pyslurm release/branch that is meant to be used for a different (and more recent) Slurm version.

For Slurm 20.2, you can use this branch though: https://github.com/PySlurm/pyslurm/tree/20.2.x

(Note that 20.2 releases do not include any of the work made in the last 1-2 years to rework the API, i.e. what is shown on pyslurm.github.io. This is because there are too many changes throughout the major slurm versions, and porting all the things back to 20.2 was too time consuming at the time.)

**Reply by HealthyPear** [reply]
I see thanks!
